### PR TITLE
Fix FHIR server URL missing trailing slash issue

### PR DESCRIPTION
**Pull request by moria97**
If the base FHIR server [Uri](https://docs.microsoft.com/en-us/dotnet/api/system.uri?view=net-6.0) has relative parts and the relative part does not end with a trailing slash  (like /api) , then the relative part will be omitted when composing a FHIR search uri. In this pull request, we always append a trailing slash to the FHIR base uri if the trailing slash is not present.

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs [hunk 1]
```diff
@@ -94,7 +95,16 @@ public sealed class FhirApiDataClient : IFhirDataClient
         /// <returns>Uri with search parameters.</returns>
         private Uri CreateSearchUri(FhirSearchParameters searchParameters)
         {
-            var baseUri = new Uri(_dataSource.FhirServerUrl);
+            // If the baseUri has relative parts (like /api),
+            // then the relative part must be terminated with a slash, (like /api/),
+            // if the relative part of baseUri is to be preserved in the constructed Uri. See https://docs.microsoft.com/en-us/dotnet/api/system.uri.-ctor?view=net-6.0
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs:100]: What this means?  maybe ```then the relative part of baseUri is to be preserved in the constructed Uri``` ?

Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs]: Thanks, I have updated the comments

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs [hunk 2]
```diff
@@ -94,7 +95,16 @@ public sealed class FhirApiDataClient : IFhirDataClient
...
+            var serverUrl = _dataSource.FhirServerUrl;
+            if (!_dataSource.FhirServerUrl.EndsWith("/"))
+            {
+                serverUrl = $"{_dataSource.FhirServerUrl}/";
+            }
+
+            var baseUri = new Uri(serverUrl);
```
Comment (QuanWanxx) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs:106]: Do we always combine the search parameters with base url? Will the customer want to search Patient resource like ```https://example.com/a/b/c/Patient```?

Comment (moria97) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs:106]: When customer searches Patient, the final query url will look like `https://example.com/a/b/c/Patient?_lastUpdated=gtxxx&ct=xxxx&_sort=_lastUpdated....`

Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs:106]: Sorry I just thought the ```baseUri``` here is the host, got it now.

**Review by BoyaWu10 — APPROVED** [review]

**Pull request merged**